src/modeler/AutoRest.Swagger/Validation/SkuModelValidation.cs

```diff
@@ -50,7 +60,7 @@ public override bool IsValid(Dictionary<string, Schema> definitions)
 
                     bool hasName = schema.Properties.Any(property =>
                         property.Key.EqualsIgnoreCase("name") &&
-                        property.Value.Type == Model.DataType.String);
+                        (property.Value.Type == Model.DataType.String || (property.Value.Type == null && evaluateIf(property.Value.Reference, definitions, Model.DataType.String))));
```

Comment:
could the condition after the '||' call a method named something like "IsReferenceOfTypeString", if that's what's being validated here?